### ISSUE TITLE
Week 1 - Getting Started

### ISSUE THREAD

**Issue by mkcode** [MLH-Fellowship/vissl#1]
a. **Dev environment:** 

- [ ] any linux machine + your choice of editor. 

b. **Setup VISSL and Run an example workflow to get familiar with git, coding environment**
- [ ] Install VISSL from source following these instructions https://github.com/facebookresearch/vissl/blob/master/INSTALL.md#Install-from-source-in-PIP-environment (remove faiss https://github.com/MLH-Fellowship/vissl/blob/master/requirements.txt#L3 by commenting or deleting the line if using PIP). Otherwise use conda https://github.com/facebookresearch/vissl/blob/master/INSTALL.md#step-1-create-conda-environment
- [ ] Run the CPU tests following https://github.com/facebookresearch/vissl/tree/master/tests#unit-tests to ensure VISSL is installed correctly
- [ ] Follow VISSL coding practices https://github.com/facebookresearch/vissl/blob/master/dev/README.md#practices-for-coding-quality to setup the environment so that every PR satisfies coding quality in VISSL

c. **PyTorch warmup with a 60min tutorial video**
- [ ] https://pytorch.org/tutorials/beginner/deep_learning_60min_blitz.html

d. **VISSL warmup: documentation and tutorials**
- [ ] Familiarize with VISSL by reading documentation https://vissl.readthedocs.io/en/latest/
- [ ] Understand VISSL config system through this tutorial https://vissl.ai/tutorials/Understanding_VISSL_Training_and_YAML_Config
- [ ] Run this Colab tutorial to further understand VISSL https://colab.research.google.com/github/facebookresearch/vissl/blob/stable/tutorials/Train_SimCLR_on_1_gpu.ipynb

e. **[Optional]: Tensorboard**
- [ ] Read about Tensorboard https://www.tensorflow.org/tensorboard
- [ ] Read the documentation https://vissl.readthedocs.io/en/latest/visualization.html for how tensorboard works with VISSL
- [ ] Run a tensorboard visualization `python tools/run_distributed_engines.py config=test/cpu_test/test_cpu_resnet_simclr config.DATA.TRAIN.DATA_SOURCES=[synthetic] config.HOOKS.TENSORBOARD_SETUP.USE_TENSORBOARD=true config.CHECKPOINT.DIR=./checkpoints`  and verify that you can see visualization on tensorboard. Follow the tensorboard documentation for how the visualization tool can be opened in your browser on a localhost.

**Comment by uhbuhb** [comment]
Environment setup tips:
https://github.com/uhbuhb/coding-docs/blob/master/new_dev_tips.md

**Comment by jeffreyksmithjr** [comment]
Noting the need to update for OSS workflow and not use FB infra.

**Comment by prigoyal** [comment]
task updated with the links!

**Comment by jeffreyksmithjr** [comment]
I sorta said this on the call, but I really like introducing the CoLab workflow early on, to make sure that the fellows have familiarity with that workflow. Great to see that there's already a good VISSL CoLab demo.

**Comment by mkcode** [comment]
Here is the invite to the MLH discord: https://discord.gg/B3sgjgDvH4

Please message `chrisewald#7243` once you are there and get you set up with
the appropriate roles. Thanks!

**Comment by uhbuhb** [comment]
visualization tool for SSL
https://www.tensorflow.org/tensorboard

**Comment by prigoyal** [comment]
added the Optional (e) Step for the tensorboard :)
